src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -166,14 +169,14 @@ def check_literal(node, dim, index, value):
     range index is valid.
 
     :param array: the node to check.
-    :type array: :py:class:`pysclone.psyir.node.array`
+    :type array: :py:class:`pysclone.psyir.node.Array`
     :param int dim: the dimension index to check.
     :param int index: the index of the range to check (0 is the \
```

Comment:
I meant to say this last time but forgot (apologies): I don't like the use of integers (0 and 1) to indicate the lower bound or upper bound. Please change this to either use an enumeration or strings. Please do this in the other routines that are written this way as well.

Reply:
I've removed it for `_check_bound_is_full_extent` but kept it for `_check_literal` as it is the index into the range node and in general we want to check whether the different range node indices (start, stop, step) are literals or not. I could change this to "start", "stop" and "step" if necessary but I don't really think it gets us much.

Reply:
OK.

Reply:
Agreed.